pilot/pkg/networking/core/v1alpha3/tracing.go

```diff
@@ -265,19 +266,52 @@ func datadogConfig(serviceName, hostname, cluster string) (*anypb.Any, error) {
 	return protoconv.MessageToAnyWithError(dc)
 }
 
-func otelConfig(serviceName, hostname, cluster string) (*anypb.Any, error) {
-	dc := &tracingcfg.OpenTelemetryConfig{
-		GrpcService: &core.GrpcService{
-			TargetSpecifier: &core.GrpcService_EnvoyGrpc_{
-				EnvoyGrpc: &core.GrpcService_EnvoyGrpc{
-					ClusterName: cluster,
-					Authority:   hostname,
+func otelConfig(serviceName, hostname, cluster string, otelProvider *meshconfig.MeshConfig_ExtensionProvider_OpenTelemetryTracingProvider) (*anypb.Any, error) {
+	var oc *tracingcfg.OpenTelemetryConfig
+
```

Comment:
oc := &tracingcfg.OpenTelemetryConfig{
ServiceName = serviceName
}

with this, we can remove the else in L283

Reply:
you mean only checking if `http` is null, and returning? Like this:

```
func otelConfig(serviceName, hostname, cluster string, otelProvider *meshconfig.MeshConfig_ExtensionProvider_OpenTelemetryTracingProvider) (*anypb.Any, error) {
	if otelProvider.GetHttp() == nil {
		return anypb.New(&tracingcfg.OpenTelemetryConfig{
			ServiceName: serviceName,
			GrpcService: &core.GrpcService{
				TargetSpecifier: &core.GrpcService_EnvoyGrpc_{
					EnvoyGrpc: &core.GrpcService_EnvoyGrpc{
						ClusterName: cluster,
						Authority:   hostname,
					},
				},
			},
		})
	}
	// configure and return http config
```